lib/education-data-processor.ts

```
@@ -0,0 +1,108 @@
+/**
+ * Server-only utilities for processing Education and Certification data,
+ * specifically handling logo fetching and placeholder generation.
+ */
+import "server-only"; // Ensure this module is never bundled for the client
+
+import type { Education, Certification, Class } from '../types/education';
+import { fetchLogo, normalizeDomain } from './logo-fetcher';
+import { assertServerOnly } from './utils/ensure-server-only'; // Import the assertion utility
+import fs from 'fs/promises';
```

Biome (1.9.4)

[error] A Node.js builtin module should be imported with the node: protocol.

Using the node: protocol is more explicit and signals that the imported module belongs to Node.js.
Unsafe fix: Add the node: protocol.


(lint/style/useNodejsImportProtocol)